### PR TITLE
Enable LLVM optimisation pipeline viewer for LDC + ASan symbolizer

### PR DESCRIPTION
**Pull request by JohanEngelen**
I am not able to test this locally myself. I've copied and modified the code from rust.ts and clang.ts.

### CODE REVIEW

File: lib/compilers/ldc.js [hunk 1]
```diff
@@ -42,6 +42,26 @@ export class LDCCompiler extends BaseCompiler {
         this.compiler.supportsIntel = true;
         this.compiler.supportsIrView = true;
         this.compiler.irArg = ['-output-ll'];
+        
+        this.compiler.supportsLLVMOptPipelineView = true;
+        this.compiler.llvmOptArg = ['--print-after-all', '--print-before-all'];
+        this.compiler.llvmOptModuleScopeArg = ['--print-module-scope'];
+        this.compiler.llvmOptNoDiscardValueNamesArg = []; // LDC does not have a flag for this yet.
+        
+        const asanSymbolizerPath = path.dirname(this.compiler.exe) + '/llvm-symbolizer';
```
Comment (partouf) [inline lib/compilers/ldc.js:51]: can you use path.join() here instead of concatting

**Comment by partouf** [comment]
thanks, we can probably test it

File: lib/compilers/ldc.js (hunk 1 above)
Comment (JohanEngelen) [reply lib/compilers/ldc.js:51]: i copied this from clang.js.  path.join(this.compiler.exe, '/llvm......') ?

Comment (partouf) [reply lib/compilers/ldc.js]: you're right, it's only in some really old version from 2017 but not the more recent ones

Comment (JohanEngelen) [reply lib/compilers/ldc.js]: Is it possible to refer to a CE-wide llvm-symbolizer that any compiler/language can use?

Comment (partouf) [reply lib/compilers/ldc.js]: Definitely, will make a change tonight to use one of the clang installations when im behind laptop

**Comment by RubenRBS** [comment]
A test is failing with an unexpected `null` value https://github.com/compiler-explorer/compiler-explorer/runs/7204244962?check_suite_focus=true#step:5:213

**Comment by partouf** [comment]
this is now live

**Pull request merged**